grouphug/model.py

```diff
@@ -206,26 +204,64 @@ def set_output_embeddings(self, new_embeddings):
             "set_output_embeddings is not implemented, use tie weights."
         )  # in resizing embeddings?
 
+    # optimization
+    def optimize(self, tokenizer=None, jit=True,ipex=None, dtype=torch.float32, level="O1"):
+        """
+        Optimizes for inference
+
+        tokenizer:
+        jit:
+        ipex: None will try
+        dtype: for ipex
+        level: for ipex
+        """
+        tokenizer = tokenizer or self.tokenizer()
+        encoded = tokenizer('example text',return_tensors="pt")
+        input_names = []
+        for name in inspect.signature(self.base_model.forward).parameters:
+            if name not in encoded:
+                break
+            input_names.append(name)
+        tuple_input = tuple(encoded[name] for name in input_names)
+        print(len(tuple_input),'inputs',input_names)
+
+        optimized_model = self.base_model.eval()
+
+        if ipex is not False:
+            ipex_available = is_ipex_available()
+            if not ipex_available:
+                if ipex:
+                    raise ImportError("Using IPEX but IPEX is not installed.")
+                else:
+                    logger.warning("IPEX not installed, so skipping this optimization")
+            else:
+                import intel_extension_for_pytorch as ipex
+                optimized_model = ipex.optimize(optimized_model, dtype=dtype, level=level)
+
+        if jit:
+            optimized_model = torch.jit.trace(optimized_model, tuple_input ,strict=False)
+            optimized_model = torch.jit.freeze(optimized_model)
+        setattr(self, self.base_model_prefix, optimized_model)
+
     # inference methods
 
     def calculate_model_embeddings(self, prefix="", **kwargs):
         encoder = self.base_model
```

Comment:
Should we use `self.base_model` or `getattr(self, self.base_model_prefix)` since the last is the one updated with `optimized_model`?

Reply:
```

    @property
    def base_model(self):
        return getattr(self, self.base_model_prefix, self)
```